### PR TITLE
fix(cli): strip URL credentials before fork detection (#59584)

### LINKED ISSUE #59584: Bug: _is_fork() false positive when origin URL contains embedded credentials
## Description

`_is_fork()` in `hermes_cli/main.py` (line 6665) does a naive string comparison of the origin remote URL against `OFFICIAL_REPO_URLS`. When the origin URL contains an embedded authentication token (e.g. `https://<token>@github.com/NousResearch/hermes-agent.git`), the comparison fails because none of the official URLs include a credentials prefix.

This produces a false "⚠ Updating from fork:" warning during `hermes update`, even though the user is pulling from the genuine NousResearch/hermes-agent repo.

## Reproduction

1. Clone the repo with a token-based URL: `git clone https://<token>@github.com/NousResearch/hermes-agent.git`
2. Run `hermes update`
3. Terminal shows "⚠ Updating from fork:" followed by the URL

## Root Cause

In `_is_fork()` (line 6665-6679):

```python
def _is_fork(origin_url: Optional[str]) -> bool:
    if not origin_url:
        return False
    normalized = origin_url.rstrip("/")
    if normalized.endswith(".git"):
        normalized = normalized[:-4]
    for official in OFFICIAL_REPO_URLS:
        official_normalized = official.rstrip("/")
        if official_normalized.endswith(".git"):
            official_normalized = official_normalized[:-4]
        if normalized == official_normalized:
            return False
    return True
```

The normalization strips `.git` and trailing `/` but does not strip the `https://<credentials>@` prefix. A URL like `https://ghp_xxxx@github.com/NousResearch/hermes-agent` normalizes to `https://ghp_xxxx@github.com/NousResearch/hermes-agent` which never matches any entry in `OFFICIAL_REPO_URLS`.

## Suggested Fix

Strip credentials from the URL before comparison. Add a step that removes the `user:token@` or `token@` portion:

```python
import re

def _strip_credentials(url: str) -> str:
    return re.sub(r"https://[^@]+@", "https://", url)
```

Then call `_strip_credentials(normalized)` before comparing against the official URLs. This handles both `https://token@github.com/...` and `https://user:token@github.com/...` patterns.

## Impact

Purely cosmetic — the update proceeds correctly regardless. But it's confusing for users who see a fork warning on a genuine installation.

### PR DESCRIPTION
**Pull request by Tranquil-Flow**
## What

`_is_fork()` in `hermes_cli/main.py` compared the origin remote URL against `OFFICIAL_REPO_URLS` with only `.git`/trailing-slash normalization. When the clone URL embeds credentials (e.g. `https://<token>@github.com/NousResearch/hermes-agent.git`), the comparison never matches and the repo is misclassified as a fork — printing a false "Updating from fork:" warning during the in-tree version check even though the user is on the genuine repo.

Fixes #59584.

## How

Extract a small `_strip_url_credentials()` helper that removes the `user:token@` / `token@` portion from any `scheme://` URL before the existing normalization, and call it in `_is_fork()`. SSH URLs (`git@github.com:...`) have no `://` scheme and are returned unchanged.

The split uses `rsplit("@", 1)` per RFC 3986, so passwords that themselves contain `@` are handled correctly (userinfo is everything before the **last** `@` before the host). The change is purely additive — the existing `OFFICIAL_REPO_URLS` comparison logic is untouched.

## Verification

- RED on `upstream/main`: 4 new tests fail — token-prefixed HTTPS URLs are reported as forks.
- GREEN after fix: all tests pass.
- The `@`-in-password edge-case test fails under a naive `split("@", 1)` and passes under `rsplit("@", 1)`.

```
tests/hermes_cli/test_fork_detection.py                            13 passed
tests/hermes_cli/test_cmd_update.py,
  test_update_check.py,
  test_update_hangup_protection.py,
  test_update_interrupted_recovery.py                              75 passed
```
Total: 88 passed, 0 regressions.

Scope: single consumer of `_is_fork` is the CLI update path; no sibling URL comparisons share the bug.

Auto-published by Moonsong via Path B automated pipeline.

### CODE REVIEW

**Review by teknium1 — COMMENTED** [review]
Thanks for isolating a real false-positive in the update path. Current HEAD still compares the raw origin string in `hermes_cli/main.py:6723-6730`, and `cmd_update()` uses that result at `hermes_cli/main.py:9663-9668`.

### Problems
- `hermes_cli/main.py:6682` splits the entire URL remainder on its last `@`, including the path. A path may contain `@`, so `https://token@github.com/someuser@github.com/NousResearch/hermes-agent.git` is reduced to the official URL and incorrectly classified as non-fork.
- The existing warning still prints raw `origin_url` at `hermes_cli/main.py:9668` for credentialed actual-fork remotes. The related discussion on #35179 identified this as a credential-exposure path.

### Suggested changes
- Restrict userinfo removal to the parsed URL authority and add a path-`@` regression test.
- Display a credential-free URL in the fork warning and test a credentialed actual-fork remote.

This is an automated hermes-sweeper review.

File: hermes_cli/main.py [hunk 1]
```diff
@@ -6662,12 +6662,35 @@ def _get_origin_url(git_cmd: list[str], cwd: Path) -> Optional[str]:
...
+    are returned unchanged.
+    """
+    if "://" not in url:
+        return url
+    scheme, rest = url.split("://", 1)
+    if "@" not in rest:
+        return url
+    # Per RFC 3986 the userinfo is everything before the LAST "@" before the
+    # host, so rsplit correctly handles passwords that contain "@".
+    credentials, host_and_path = rest.rsplit("@", 1)
```
Comment (teknium1) [inline hermes_cli/main.py]: `rest` includes path/query/fragment, where `@` is valid. Restrict the split to the parsed authority (or use `urlsplit`) before reconstructing the URL; otherwise a fork URL such as `https://token@github.com/someuser@github.com/NousResearch/hermes-agent.git` reduces to the official URL and is misclassified.

**Comment by Tranquil-Flow** [comment]
Addressed in `8f5bd2b3`:

- Parse the URL authority with `urlsplit()` and remove userinfo only from `netloc`, preserving `@` in paths, queries, and fragments.
- Sanitize the credentialed fork URL before printing the update warning.
- Add regressions for a path containing `@` and the production warning path for a credentialed actual fork.

Verification: `test_fork_detection.py` (15 passed), `test_update_autostash.py` (33 passed), and Ruff (passed). Branch is rebased onto current `main`.
